docs/src/main/asciidoc/jfr.adoc

```diff
@@ -0,0 +1,286 @@
+We can use the jfr CLI to see a list of events limited to those related to Quarkus in the dump file by doing the following.
+
+----
+jfr print --categories quarkus myrecording.jfr
+----
+
+=== JDK Mission Control
+
+JMC is essentially a GUI viewer for JFR.
+Some distributions include JMC in OpenJDK binary, but if not, we need to download it manually.
+To see a list of events using the JMC, first we load the JFR file in the JMC as follows.
+
+----
+jmc -open myrecording.jfr
+----
+
+After opening the JFR file, we have two options.
+One is to view the events as a tabular list, and the other is to view the events on the threads in which they occurred, in chronological order.
+
+To view Quarkus events in tabular style, select the Event Browser on the left side of the JMC, then open the Quarkus event type tree on the right side of the JMC.
+
+image::jfr-event-browser.png[alt=JDK Mission Control Event Browser view,role="center"]
+
+To see Quarkus events in chronological order on a thread, select the `Java application` and `Threads` on the left side of the JMC.
+
+image::jfr-java-ap-thread.png[alt=JDK Mission Control thread view,role="center"]
+
+The standard configuration does not show Quarkus events.
+We have to do three tasks to see the Quarkus events.
+
+1. Right-click and select `Edit Thread Activity Lanes...`.
+2. Select the plus button to add a new lane on the left side, then check to display that lane.
+3. Select Quarkus as the event type that lane will display and press OK.
+
+image::jfr-edit-thread-activity-lanes.png[alt=JDK Mission Control Edit Thread Activity Lanes,role="center"]
+
+Now we can see the Quarkus events per thread.
+
+image::jfr-thread.png[alt=JDK Mission Control thread view,role="center"]
+
+[NOTE]
+====
+Non-blocking is where multiple processes are processed apparently simultaneously in a single thread.
+Therefore, this extension records multiple JFR events concurrently, and a number of events might overlap on the JMC.
+This could make it difficult for you to see the events you want to see.
+To avoid this, we recommend to use Request IDs (described below) to filter events so that you only see the information about the requests you want to see.
+====
+
+== Events
+
+=== Identifying Requests
+This extension works with the OpenTelemetry extension.
+The events recorded by this extension have a trace ID and a span ID. These are recorded with the OpenTelemetry IDs respectively.
+
+This means that after we identify the trace and span IDs of interest from the UI provided by the OpenTelemetry implementation, we can immediately jump to the details in JFR using those IDs.
+
+If we have not enabled the OpenTelemetry extension, this extension creates an ID for each request and links it to JFR events as a traceId.
+In this case, the span ID will be null.
+
+For now, Quarkus only has REST events, but we plan to use this ID to link each event to each other as we add more events in the future.
```

Comment:
is there some docs on how/what  extension writers can do to add more events to jfr ?

Reply:
I have added content for exstension writers to the document.